src/rgw/rgw_sal_rados.cc

```diff
@@ -2478,6 +2493,21 @@ int RadosMultipartUpload::get_info(const DoutPrefixProvider *dpp, optional_yield
   return 0;
 }
 
+std::unique_ptr<Writer> RadosMultipartUpload::get_writer(
+				  const DoutPrefixProvider *dpp,
+				  optional_yield y,
+				  std::unique_ptr<rgw::sal::Object> _head_obj,
+				  const rgw_user& owner, RGWObjectCtx& obj_ctx,
+				  const rgw_placement_rule *ptail_placement_rule,
+				  uint64_t part_num,
+				  const std::string& part_num_str)
+{
+  auto aio = rgw::make_throttle(store->ctx()->_conf->rgw_put_obj_min_window_size, y);
+  return std::unique_ptr<Writer>(new RadosMultipartWriter(dpp, y, this,
```

Comment:
you can still use `std::make_unique<RadosMultipartWriter>()` even though the return type is `std::unique_ptr<Writer>` - unique_ptr supports the same implicit conversions as the underlying pointer

Reply:
This is the style used everywhere in rgw_sal_rados.cc.  I'll add to my list to fix them all in a style patch.